server/consumer.go

```diff
@@ -3282,20 +3281,18 @@ func (o *consumer) getNextMsg() (*jsPubMsg, uint64, error) {
 			o.mu.Unlock()
 			pmsg := getJSPubMsgFromPool()
 			sm, sseq, err := store.LoadNextMsg(filterSubject, filterWC, nextSeq, &pmsg.StoreMsg)
-			o.mu.Lock()
 
 			filter.err = err
```

Comment:
Can these be manipulated safely without the consumer lock?

Reply:
During normal consumer operations, yes.
Added a test for that.

However... I think it's a good point and it's safer to move it before manipulating filter fields - especially since this could surface bugs after some changes to consumers, their updates etc. 

Thx!

Reply:
Why are you unlocking and locking in the first place? To not hold the lock during load?

Reply:
The locking and unlocking in consumer was added here: https://github.com/nats-io/nats-server/pull/3922/files, specifically in commit https://github.com/nats-io/nats-server/pull/3922/commits/13167f46b9884f3bb4b647a49f928c02af6fb325 as a performance improvement under heavy load.